### PR TITLE
robust function to generate file name according to IMOS conventions

### PR DESCRIPTION
**Pull request by diodon**
some source file names does not comply with IMOS file naming conventions. platform code is set to "VARIOUS" as it could be more than one, depending on the input files to aggregate

### CODE REVIEW

**Comment by diodon** [comment]
Do not merge yet. I'm reviewing some results in TPAC...

**Comment by diodon** [comment]
now it's working. good to go.

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py [hunk 1]
```diff
@@ -179,29 +179,66 @@ def set_variableattr(varlist, variable_attribute_dictionary, add_variable_attrib
     return variable_attributes
 
 
-def generate_netcdf_output_filename(fileURL, nc, VoI, file_product_type, file_version):
+def get_data_code(VoI):
+    """
+    get data code sensu IMOS conventions from variable code
```
Comment (mhidas) [inline ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: "sensu"??

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py [hunk 2]
```diff
@@ -179,29 +179,66 @@ def set_variableattr(varlist, variable_attribute_dictionary, add_variable_attrib
...
+    return dataCodes[VoI]
+
+def get_facility_code(fileURL):
+    """
+    get the facility code from the file URL
+
+    :param fileURL: URL of a file
+    :return: facility code
+    """
+    split_filename = fileURL.split("/")[-1].split("_")
```
Comment (mhidas) [inline ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: `os.path.basename(fileURL)` would be a bit more robust than `fileURL.split("/")[-1]`

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py [hunk 3]
```diff
@@ -179,29 +179,66 @@ def set_variableattr(varlist, variable_attribute_dictionary, add_variable_attrib
...
         VoI = VoI.replace('_', '-')
     t_start = pd.to_datetime(nc.TIME.min().values).strftime(file_timeformat)
     t_end = pd.to_datetime(nc.TIME.max().values).strftime(file_timeformat)
-    split_path = fileURL.split("/")
-    split_parts = split_path[-1].split("_") # get the last path item (the file nanme)
 
-    output_name = '_'.join([split_parts[0] + "_" + split_parts[1] + "_" + split_parts[2], \
-                            t_start, split_parts[4], "FV0" + str(file_version), (VoI+"-"+file_product_type)]) + \
-                            "_END-" + t_end + "_C-" + datetime.utcnow().strftime(file_timeformat) + ".nc"
+    output_name = '_'.join(['IMOS', facility_code, data_code, t_start, site_code, ('FV0'+str(file_version)), (VoI+"-"+product_type), t_end, 'C-' + datetime.utcnow().strftime(file_timeformat)]) + '.nc'
```
Comment (mhidas) [inline ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: Need to add "END-" in front of `t_end`

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py (hunk 1 above)
Comment (diodon) [reply ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: https://en.wikipedia.org/wiki/Sensu

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py (hunk 3 above)
Comment (diodon) [reply ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: added

File: ANMN/LTSP/TSaggregator/aggregated_timeseries.py (hunk 2 above)
Comment (diodon) [reply ANMN/LTSP/TSaggregator/aggregated_timeseries.py]: ok.

**Comment by mhidas** [comment]
:+1: Looks good now!

**Pull request merged**